### DISCUSSION TITLE
Use devnet in CI/CD exclusively for develop merges and releases

### DISCUSSION THREAD

**Discussion by ivpavici** [seanjameshan/starknet.js#386] [Ideas]
### Problem: 
- Testnet is very slow ATM
- Regardless, even before, running tests on each merge on the testnet takes around 1.5 - 2 hours
- Running tests on testnet requires adding funds from the faucet on accounts

### Proposal:
I would like to propose a change in Starknet.js CI/CD flow and the proposal involves **replacing testnet with devnet for merges in develop branch** -> that means, for pre-releases that users can get with the `npm install starknet@next` command

Merges in `main` branch would stay the same (run on testnet).

### Reasoning:
- Faster!
- Devnet should work the same way as testnet (if not, it's a bug and we must fix it anyway)
- Some projects have abandoned the testnet for devent (for example Braavos) and offered devnet in the cloud -> this is an indicator that devnet can be used for this purpose
- Testnet would still be used for an **official release** 

**Note**:
We can emphasize more clearly that `npm install starknet@next` is a pre-release, which ran on devnet

**Comment by yoga-braavos** [comment]
Fully agree that `@next` can run solely on devnet

**Comment by dhruvkelawala** [comment]
Then we should also add devnet-RPC tests for pre-release, no? Seems like we will all be moving the RPC asap.

**Reply by ivpavici** [reply]
Yeah, devnet supports RPC 0.2.1 : https://github.com/Shard-Labs/starknet-devnet/pull/332

**Comment by tabaktoni** [comment]
Resolved
